libraries/AP_Compass/AP_Compass.cpp

```diff
@@ -620,14 +620,21 @@ void Compass::_detect_backends(void)
         ADD_BACKEND(DRIVER_AK09916, AP_Compass_AK09916::probe(*this, hal.i2c_mgr->get_device(0, HAL_COMPASS_AK09916_I2C_ADDR),
                                                               both_i2c_external, both_i2c_external?ROTATION_YAW_270:ROTATION_NONE),
                     AP_Compass_AK09916::name, both_i2c_external);
+
+        // IST8310 on external and internal bus
+        ADD_BACKEND(DRIVER_IST8310, AP_Compass_IST8310::probe(*this, hal.i2c_mgr->get_device(1, HAL_COMPASS_IST8310_I2C_ADDR),
+                                                              true, ROTATION_PITCH_180), AP_Compass_IST8310::name, true);
```

Comment:
The last argument `external` doesn't do anything and should rather be removed.

Or... the `force_external` argument be removed from probe() and let _add_backend() do the right thing in case probe succeeded.  In that case what you are trying to do here would be a given, for any backend.

Reply:
Just to add that `name` doesn't do anything either. Both were added at a time we printed every backend that was added. That was since removed so they aren't used anymore.

Reply:
removing that external argument should be a separate patch from the IST8310 support. It applies to all drivers.